extensions/core/deployment/src/main/java/io/quarkiverse/cxf/deployment/QuarkusCxfProcessor.java

```diff
@@ -211,8 +211,10 @@ void buildXmlResources(BuildProducer<NativeImageResourceBuildItem> resources,
         try (ByteArrayOutputStream os = new ByteArrayOutputStream()) {
             XPath xpath = XPathFactory.newInstance().newXPath();
             DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
-            documentBuilderFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "");
-            documentBuilderFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
+            documentBuilderFactory.setFeature(XMLConstants.FEATURE_SECURE_PROCESSING, true);
+            String originalDTDProperty = System.getProperty("javax.xml.accessExternalDTD");
+            System.setProperty("javax.xml.accessExternalDTD", "file, http");
```

Comment:
I wonder whether we really need to set the system property for this to work? Wouldn't `documentBuilderFactory.setAttribute(XMLConstants.ACCESS_EXTERNAL_DTD, "file,http");` have the same effect?

I am asking because setting the system property might lead to race conditions: the BuildStep methods are run in parallel and there might exist other concurrent BuildStep method setting the same property.

Reply:
Excellent point @ppalaga !  `documentBuilderFactory.setAttribute` should work fine, let me address this.